src/creature.cpp

```diff
@@ -1772,10 +1772,8 @@ std::vector<effect> Creature::get_effects_from_bp( const bodypart_id &bp ) const
 {
     std::vector<effect> effs;
     for( auto &elem : *effects ) {
-        for( const std::pair<const bodypart_id, effect> &_it : elem.second ) {
-            if( _it.first == bp ) {
-                effs.push_back( _it.second );
-            }
+        if( elem.second.find( bp ) != elem.second.end() ) {
+            effs.push_back( elem.second[bp] );
```

Comment:
If this affects perf you can shave off another lookup for "free"
```suggestion
        const auto iter = elem.second.find( bp );
        if( iter != elem.second.end() ) {
            effs.push_back( iter->second );
```

returning pointers or std::reference_wrapper instead of copying over `effect`s is also an option if callers can be modified to use em

Reply:
Thanks!